repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -118,6 +120,27 @@ private void processUpdateProduct(AtlasEntity entity, AtlasVertex vertex) throws
             throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "DataProduct can only be moved to another Domain.");
         }
 
+        AtlasEntity diffEntity = RequestContext.get().getDifferentialEntity(entity.getGuid());
+        String state = vertex.getProperty(STATE_PROPERTY_KEY, String.class);
+
+        if(entity.getAttribute(DAAP_LINEAGE_STATUS_ATTR) != null && entity.getAttribute(DAAP_LINEAGE_STATUS_ATTR).equals(DAAP_LINEAGE_STATUS_COMPLETED)){
+            if (!ARGO_SERVICE_USER_NAME.equals(RequestContext.getCurrentUser())) {
+                throw new AtlasBaseException(AtlasErrorCode.UNAUTHORIZED_ACCESS, RequestContext.getCurrentUser(), "Can not update Lineage Status to Completed");
+            }
+
+            if (vertex.getProperty(DAAP_LINEAGE_STATUS_ATTR, String.class).equals(DAAP_LINEAGE_STATUS_IN_PROGRESS)){
+                entity.setAttribute(DAAP_LINEAGE_STATUS_ATTR, DAAP_LINEAGE_STATUS_COMPLETED);
+            }
+        }
+
+        if(entity.hasAttribute(DAAP_ASSET_DSL_ATTR) && diffEntity.hasAttribute(DAAP_ASSET_DSL_ATTR)){
```

Comment:
Dont't we use attribute `dataProductAssetsPlaybookFilter` ?

Reply:
1. https://atlanhq.slack.com/archives/C05NR8V0RE3/p1741351430978409?thread_ts=1740939963.303599&cid=C05NR8V0RE3
2. https://atlanhq.slack.com/archives/C05NR8V0RE3/p1741352452910059?thread_ts=1740939963.303599&cid=C05NR8V0RE3